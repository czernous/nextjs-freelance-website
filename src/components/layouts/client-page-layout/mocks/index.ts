import { IClientPageLayoutProps } from '../../../../interfaces';

export const clientPageMock: IClientPageLayoutProps = {
  pageTitle: 'About',
  meta: {
    metaDescription:
      'This is a dummy description designed for testing purposes',
    metaKeywords: 'some meta keywords,testing,typescript,blog',
    openGraph: {
      title: 'This title will be shown on social media',
      description: 'This description will be shown on social media',
      imageUrl: '',
      type: '',
      url: '',
    },
  },
};
