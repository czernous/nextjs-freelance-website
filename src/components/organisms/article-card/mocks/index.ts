import { IArticleCard, IImage } from '../../../../interfaces';
import { base64ImgSmall, base64ImgMedium, base64ImgLarge } from './images.mock';

const imageMock: IImage = {
  publicId: 'fakeurl.com/asd',
  version: 1,
  signature: 'fakeSignature',
  width: 1000,
  height: 666,
  format: 'jpg',
  resourceType: 'fake resource',
  bytes: 111,
  type: '',
  url: base64ImgLarge,
  secureUrl: base64ImgLarge,
  responsiveUrls: [
    { width: 1000, url: base64ImgLarge },
    { width: 700, url: base64ImgMedium },
    { width: 500, url: base64ImgSmall },
  ],
  thumbnailUrl: base64ImgSmall,
  blurredImageUrl: base64ImgSmall,
  path: 'fakeurl',
  id: {
    timestamp: 111,
    machine: 111,
    pid: -999,
    increment: 111,
    creationTime: '2022-06-25T15:44:40.411Z',
  },
  _id: '12345',
  createdAt: '2022-06-25T15:44:40.411Z',
  updatedAt: '2022-06-25T15:44:40.411Z',
  altText: 'mock image',
};

export const cardMock: IArticleCard = {
  title: 'Article title',
  unoptimized: true,
  description:
    'Some random article short summary. This will be taken from the article itself or prefilled when the article is posted',
  ctaText: 'Read more',
  ctaUrl: 'https//google.com',
  image: imageMock,
};
