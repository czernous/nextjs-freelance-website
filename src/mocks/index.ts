export * from './page-data';
export * from './posts';
export * from './paginated-posts';
