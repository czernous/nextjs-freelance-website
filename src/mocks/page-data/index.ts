import { IHomePage, IServicesPage } from '@src/interfaces';

export const homePageMock: IHomePage = {
  pageFields: {
    ctaBtnHref: 'https://google.com',
    ctaBtnText: 'Get stuff',
    ctaHeadline: 'Are you looking to get awesome stuff??',
    ctaSubheadline: 'We have the most awesome stuff on the internet!',
  },
  slug: 'home',
  meta: {
    metaDescription: 'Test',
    metaKeywords: 'test',
    openGraph: {
      title: 'Test',
      description: 'test',
      imageUrl: 'http://g.com',
    },
  },

  updatedAt: '2023-02-16T15:47:25.559Z',
};

export const servicesPageMock: IServicesPage = {
  pageFields: {
    content:
      '<h2>Service #1</h2><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis nunc libero. Nullam tincidunt lectus ut maximus dignissim. Mauris et risus in metus ornare convallis. Nullam in orci leo. Aenean sit amet arcu vitae ipsum mollis blandit. Pellentesque pellentesque sapien mollis, pharetra magna sit amet, varius eros. Donec eu dui ullamcorper ligula rutrum tincidunt. Sed eu venenatis mi. Nullam rutrum sagittis tortor, id feugiat dolor commodo a. Nullam venenatis rutrum elementum. Vestibulum vehicula lectus in nisl porta scelerisque at id metus. Sed porttitor purus eu cursus porta. Donec suscipit semper ligula, ut eleifend nunc commodo ut. Aenean sed mauris molestie, gravida nisi et, lobortis sapien. Nullam et massa mi. </p> <p>Nulla a sem sit amet nisl interdum tincidunt. Fusce pellentesque nec risus et lacinia. Donec dignissim enim ac vehicula malesuada. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam at varius augue. Phasellus varius, sem ac posuere mattis, augue turpis sollicitudin arcu, ut tincidunt arcu tellus vel sapien. Nulla sagittis auctor velit, ut tristique nulla sollicitudin et. Duis ac purus efficitur, congue purus egestas, ultricies quam. Nulla tincidunt nibh risus, vel semper lacus consectetur quis. Aenean luctus enim in justo rhoncus dignissim. Donec pellentesque fermentum sapien vitae aliquam.</p><h2>Service #1</h2><p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Curabitur quis nunc libero. Nullam tincidunt lectus ut maximus dignissim. Mauris et risus in metus ornare convallis. Nullam in orci leo. Aenean sit amet arcu vitae ipsum mollis blandit. Pellentesque pellentesque sapien mollis, pharetra magna sit amet, varius eros. Donec eu dui ullamcorper ligula rutrum tincidunt. Sed eu venenatis mi. Nullam rutrum sagittis tortor, id feugiat dolor commodo a. Nullam venenatis rutrum elementum. Vestibulum vehicula lectus in nisl porta scelerisque at id metus. Sed porttitor purus eu cursus porta. Donec suscipit semper ligula, ut eleifend nunc commodo ut. Aenean sed mauris molestie, gravida nisi et, lobortis sapien. Nullam et massa mi. </p> <p>Nulla a sem sit amet nisl interdum tincidunt. Fusce pellentesque nec risus et lacinia. Donec dignissim enim ac vehicula malesuada. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam at varius augue. Phasellus varius, sem ac posuere mattis, augue turpis sollicitudin arcu, ut tincidunt arcu tellus vel sapien. Nulla sagittis auctor velit, ut tristique nulla sollicitudin et. Duis ac purus efficitur, congue purus egestas, ultricies quam. Nulla tincidunt nibh risus, vel semper lacus consectetur quis. Aenean luctus enim in justo rhoncus dignissim. Donec pellentesque fermentum sapien vitae aliquam.</p>',
  },
  slug: 'services',
  meta: {
    metaDescription: 'string',
    metaKeywords: 'string',
    openGraph: {
      title: 'string',
      description: 'string',
      imageUrl: 'string',
    },
  },

  updatedAt: '2023-02-16T07:05:35.214Z',
};
