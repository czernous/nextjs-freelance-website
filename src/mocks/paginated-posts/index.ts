import { posts } from '../posts';

export const mockPagedData = {
  data: posts.slice(0, 10),
  hasPagination: true,
  page: 1,
  pageSize: 10,
  totalPages: Math.ceil(posts.length / 10),
  totalDocuments: posts.length,
};
