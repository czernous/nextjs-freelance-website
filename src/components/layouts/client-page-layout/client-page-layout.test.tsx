import { render, screen, waitFor } from '@testing-library/react';
import '@testing-library/jest-dom';
import ClientPageLayout from './index';
import { clientPageMock } from './mocks';
import { axe, toHaveNoViolations } from 'jest-axe';

import React from 'react';

expect.extend(toHaveNoViolations);

// jest.mock('next/config', () => () => ({
//   serverRuntimeConfig: {
//     PROJECT_ROOT: __dirname,
//   },
//   publicRuntimeConfig: {
//     APP_NAME: 'test',
//   },
// }));

describe('Client Page Layout', () => {
  const template = (
    <ClientPageLayout
      test-id="layout"
      pageTitle={clientPageMock.pageTitle}
      meta={clientPageMock.meta}
    />
  );

  it('passes axe audit', async () => {
    const { container } = render(template);
    await waitFor(async () =>
      expect(await axe(container)).toHaveNoViolations(),
    );
  });

  it('is present in layout', async () => {
    render(template);

    const layout = await screen.findByTestId('client-layout');
    await waitFor(() => {
      expect(layout).toBeInTheDocument();
    });
  });

  it('is rendered unchanged', async () => {
    const { container } = render(template);
    expect(container).toMatchSnapshot();
  });
});
