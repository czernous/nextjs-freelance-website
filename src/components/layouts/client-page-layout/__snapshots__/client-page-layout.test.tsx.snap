// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client Page Layout is rendered unchanged 1`] = `
<div>
  <div
    data-testid="client-layout"
  >
    <div
      class="wrapper"
    >
      <nav
        aria-label="Main"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-3y7qso-MuiPaper-root-MuiAppBar-root"
        role="navigation"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular custom-container css-hyum1k-MuiToolbar-root"
          style="width: 100%; height: 88px;"
        >
          <button
            aria-label="open drawer"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-6iz4mj-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            test-id="drawerButton"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="MuiTypography-root MuiTypography-h6 css-1urdvp-MuiTypography-root"
          >
            test
          </div>
          <div
            class="MuiBox-root css-130f8nx"
          >
            <a
              href="/"
              style="text-decoration: none;"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-10s16q3-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Home
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
            <a
              href="/about"
              style="text-decoration: none;"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-12b7yuv-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                About
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
            <a
              href="/blog"
              style="text-decoration: none;"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-12b7yuv-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Blog
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
            <a
              href="/services"
              style="text-decoration: none;"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-12b7yuv-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Services
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
            <a
              href="/contact"
              style="text-decoration: none;"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-12b7yuv-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Contact
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
        </div>
      </nav>
      <nav
        class="MuiBox-root css-0"
      />
      <main
        class="layout custom-container"
      >
        <h1
          class="heading"
        >
          About
        </h1>
        <section
          class="content"
        />
         
      </main>
    </div>
  </div>
</div>
`;
